core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -241,36 +245,86 @@ FuturePtr<cosmos::TransactionsBulk> GaiaCosmosLikeBlockchainExplorer::getTransac
                 for (const auto &node : transactions) {
                     auto tx = cosmos::Transaction();
                     parseTransactionWithPosttreatment(node, tx);
-                    result->transactions.emplace_back(tx);
+                    result.transactions.emplace_back(tx);
                 }
 
                 if (!document.HasMember(kCount) ||
                     !document[kCount].IsString() ||
                     !document.HasMember(kTotalCount) ||
                     !document[kTotalCount].IsString()) {
-                    result->hasNext = false;
+                    result.hasNext = false;
                 }
                 else {
                     const auto count = std::stoi(document[cosmos::constants::kCount].GetString());
                     const auto total_count =
                         std::stoi(document[kTotalCount].GetString());
-                    result->hasNext = (count < total_count);
+                    result.hasNext = (count < total_count);
                 }
-                return result;
-            });
+                return Future<cosmos::TransactionsBulk>::successful(result);
+                })
+                .flatMapPtr<cosmos::TransactionsBulk>(
+                getContext(),
+                [this](const cosmos::TransactionsBulk& inputBulk) mutable -> FuturePtr<cosmos::TransactionsBulk> {
+                    auto retval = inputBulk;
+                    auto inputTxs = inputBulk.transactions;
```

Comment:
The copy is necessary here ? If I understand correctly that variable is more an variable alias than something else.

```suggestion
                    auto const &inputTxs = inputBulk.transactions;
```

Reply:
Yes will do ; I used to have code invalidating that reference closed to it earlier, I think that's why I had a copy, and forgot to change it after